From patchwork Tue Sep  5 18:20:09 2017
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: [U-Boot] env: Fix out of tree building of tools-all
X-Patchwork-Submitter: Tom Rini <trini@konsulko.com>
X-Patchwork-Id: 810244
Message-Id: <1504635609-23585-1-git-send-email-trini@konsulko.com>
To: u-boot@lists.denx.de
Date: Tue,  5 Sep 2017 14:20:09 -0400
From: Tom Rini <trini@konsulko.com>
List-Id: U-Boot discussion <u-boot.lists.denx.de>

With the move of environment code from common/ to env/ a number of
changes needed to be made to various make targets.  We missed updating
some of the files required for out of tree builds of the tools.  Correct
the 'environ' target to know that we need to work under tools/env/ still
(not tools/environ/) and then update the wrappers in env_attr.c and
env_flags.c to point to the new correct file.

Reported-by: Peter Robinson <pbrobinson@gmail.com>
Signed-off-by: Tom Rini <trini@konsulko.com>
---
 Makefile              | 2 +-
 tools/env/env_attr.c  | 2 +-
 tools/env/env_flags.c | 2 +-
 tools/env/fw_env.c    | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 792162521155..020ebd50ad22 100644
--- a/Makefile
+++ b/Makefile
@@ -1466,7 +1466,7 @@ checkarmreloc: u-boot
 	fi
 
 environ: scripts_basic
-	$(Q)$(MAKE) $(build)=tools/$@
+	$(Q)$(MAKE) $(build)=tools/env
 
 tools-only: scripts_basic $(version_h) $(timestamp_h)
 	$(Q)$(MAKE) $(build)=tools
diff --git a/tools/env/env_attr.c b/tools/env/env_attr.c
index 502d4c900bf2..4d8536335c30 100644
--- a/tools/env/env_attr.c
+++ b/tools/env/env_attr.c
@@ -1 +1 @@
-#include "../../common/env_attr.c"
+#include "../../env/attr.c"
diff --git a/tools/env/env_flags.c b/tools/env/env_flags.c
index b261cb8e3907..71e13e2021f6 100644
--- a/tools/env/env_flags.c
+++ b/tools/env/env_flags.c
@@ -1 +1 @@
-#include "../../common/env_flags.c"
+#include "../../env/flags.c"
diff --git a/tools/env/fw_env.c b/tools/env/fw_env.c
index 965e1662d702..ab06415898c2 100644
--- a/tools/env/fw_env.c
+++ b/tools/env/fw_env.c
@@ -755,7 +755,7 @@ static int flash_read_buf (int dev, int fd, void *buf, size_t count,
 
 		/*
 		 * If a block is bad, we retry in the next block at the same
-		 * offset - see common/env_nand.c::writeenv()
+		 * offset - see env/nand.c::writeenv()
 		 */
 		lseek (fd, blockstart + block_seek, SEEK_SET);
 
